### PR TITLE
Fix alignment of atomics in shared memory communication channel

### PR DESCRIPTION
**Pull request by phil-opp**
Fixes https://github.com/dora-rs/dora/issues/610

### CODE REVIEW

File: libraries/shared-memory-server/src/channel.rs [hunk 1]
```diff
@@ -188,11 +189,31 @@ impl ShmemChannel {
...
+        data as usize - base,
+    )
+}
+
+fn offset_ptrs(next_free: *mut u8) -> (*mut AtomicBool, *mut AtomicU64, *mut u8) {
+    let disconnect_ptr = next_free.wrapping_add(next_free.align_offset(align_of::<AtomicBool>()));
+    let len_ptr = disconnect_ptr
+        .wrapping_add(mem::size_of::<AtomicBool>())
+        .wrapping_add(next_free.align_offset(align_of::<AtomicU64>()));
+    let data_ptr = len_ptr.wrapping_add(mem::size_of::<AtomicU64>());
```
Comment (taoliv) [inline libraries/shared-memory-server/src/channel.rs:215]: data_ptr does not need to be aligned.

File: libraries/shared-memory-server/src/channel.rs [hunk 2]
```diff
@@ -188,11 +189,31 @@ impl ShmemChannel {
...
+    (
+        disconnect as usize - base,
+        len as usize - base,
+        data as usize - base,
+    )
+}
+
+fn offset_ptrs(next_free: *mut u8) -> (*mut AtomicBool, *mut AtomicU64, *mut u8) {
+    let disconnect_ptr = next_free.wrapping_add(next_free.align_offset(align_of::<AtomicBool>()));
+    let len_ptr = disconnect_ptr
```
Comment (taoliv) [inline libraries/shared-memory-server/src/channel.rs:212]: len_ptr is still misaligned.

The following code works for me:
```rust
    let len_ptr = disconnect_ptr
        .wrapping_add(mem::size_of::<AtomicBool>());
    let len_ptr = len_ptr
        .wrapping_add(len_ptr.align_offset(align_of::<AtomicU64>()));
```

Comment (phil-opp) [reply libraries/shared-memory-server/src/channel.rs]: Ah yes, of course. Thanks!

**Review by haixuanTao — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**